config-common/src/watcher.rs

```diff
@@ -55,10 +43,10 @@ pub fn start_watcher<TConfig: 'static, TError: 'static>(
             // Add configuration paths to be watched.
             file_watcher
                 .watch(config_directory_path, notify::RecursiveMode::Recursive)
```

Comment:
This should be `NonRecursive`.

https://docs.rs/notify/4.0.15/notify/trait.Watcher.html#tymethod.watch

>If the `path` is a directory, `recursive_mode` will be evaluated. If `recursive_mode` is `RecursiveMode::Recursive` events will be delivered for all files in that tree. Otherwise only the directory and its immediate children will be watched.

We only look for config files in the immediate children.

Reply:
Fixed.